File: strings/isPalindrome/isPalindrome.py
#  O(n) time O(1) space
def isPalindrome(string):
    startPointer = 0
    endPointer = len(string) - 1

    while startPointer < endPointer:
        if string[startPointer] != string[endPointer]:
            return False
        startPointer += 1
        endPointer -= 1
    return True


# Example usage:
if __name__ == "__main__":
    testString = "racecar"
    print(f"Is '{testString}' a palindrome? {isPalindrome(testString)}")

    testString2 = "hello"
    print(f"Is '{testString2}' a palindrome? {isPalindrome(testString2)}")

File: strings/isPalindrome/isPalindrome.js
// O(n) time O(1) space
function isPalindrome(string) {
  let startPointer = 0
  let endPointer = string.length - 1
  while(startPointer < endPointer) {
    if(string[startPointer] !== string[endPointer]) {
      return false
    }
    startPointer++
    endPointer--
  }
  return true
}

// Example usage:
console.log(isPalindrome("racecar")) // true
console.log(isPalindrome("hello")) // false